Logcall 9.0

treveradams opened the pull request:
This module can automatically log calls, including a recording to go with the call. There was a previous pull request when I renamed this from logcall to logcall-9.0. The rename was because I may do a version for Odoo 8.0.

treveradams committed:
Initial port of FreeSWITCH functionality to Odoo 9.

treveradams committed:
Further cleanups and corrections.

treveradams committed:
Logcall and related modules.

treveradams committed:
Work to possibly get recording logging working with fusionpbx.

treveradams committed:
Attempts to better isolate changes.

treveradams committed:
Fix several bugs.
Refactor and cleanup code.

treveradams committed:
Correct some documentation of scripts.
Handle channel types that have src or dst as u:{UUID}.
Handle ORIGINATOR_CANCEL.

treveradams committed:
Thank you to Alexandr Usov and Stanislav Sinyagin for testing and ideas.
* Fix originate to handle Unicode via UTF-8.
* Fix Open Caller to handle more situations and channel types.
* get_caller_name.py and pop-up functionality now handle invalid numbers properly.
* Other fixes that were already committed.

elicoidal commented on README.md:
```diff
@@ -29,6 +29,7 @@ addon | version | summary
 [base_phone_popup](base_phone_popup/) | 9.0.1.0.0 | Pop-up the related form view to the user on incoming calls
 [crm_phone](crm_phone/) | 9.0.1.0.0 | Validate phone numbers in CRM
 [event_phone](event_phone/) | 9.0.1.0.0 | Validate phone numbers in Events
+[freeswitch_click2dial](freeswitch_click2dial/) | 9.0.1.0.0 | FreeSWITCH-Odoo telephony connector (click2dial)
```
Please do not change this part: this is automatically generated and would trigger conflicts

elicoidal commented on asterisk_logcall/README.rst:
```diff
@@ -0,0 +1,6 @@
+Asterisk Phone Log-call
```
@treveradams   Thanks for your contribution to the OCA!
Please check the Travis output as there might errors/warnings to be corrected
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code..
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

elicoidal commented on asterisk_logcall/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
```
short headers

elicoidal commented on asterisk_logcall/README.rst:
```diff
@@ -0,0 +1,6 @@
+Asterisk Phone Log-call
+=========================
+
+This bridge adds an option to the Asterisk IPBX servers for the URL to pull
```
Please use README template

elicoidal commented on asterisk_logcall/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+
+{
+    'name': 'Asterisk Phone Log-call',
+    'version': '9.0.0.1.0',
```
9.0.1.0.0

elicoidal commented on asterisk_logcall/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+    'category': 'Phone',
+    'license': 'AGPL-3',
+    'summary': 'Necessary bridge between log-call and Asterisk Click2dial',
+    'author': "Trever L. Adams",
```
If you are part of a company you should input the company here (and individual contributor in the README, along with email)

treveradams replied:
On 08/25/2016 05:44 PM, Eric Caudal wrote:

> In asterisk_logcall/**openerp**.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76345008:
> 
> > +#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +#    GNU Affero General Public License for more details.
> > +#
> > +#    You should have received a copy of the GNU Affero General Public License
> > +#    along with this program.  If not, see http://www.gnu.org/licenses/.
> > +#
> > +##############################################################################
> > +
> > +
> > +{
> > -    'name': 'Asterisk Phone Log-call',
> > -    'version': '9.0.0.1.0',
> > -    'category': 'Phone',
> > -    'license': 'AGPL-3',
> > -    'summary': 'Necessary bridge between log-call and Asterisk Click2dial',
> > -    'author': "Trever L. Adams",
> 
> If you are part of a company you should input the company here (and
> individual contributor in the README, along with email)
> 
> Is email address required? I am contributing as an individual.

elicoidal replied:
No, this is just a suggestion but you might want to add an email in the 
README in case at some point there is some claim on copyright. If we 
cannot join the authors ("best effort"), it makes things a little bit 
more tricky.

Again this is a suggestion.

Eric CAUDAL

treveradams replied:
On 08/26/2016 01:16 AM, Eric Caudal wrote:

> In asterisk_logcall/**openerp**.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76373869:
> 
> > +#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +#    GNU Affero General Public License for more details.
> > +#
> > +#    You should have received a copy of the GNU Affero General Public License
> > +#    along with this program.  If not, see http://www.gnu.org/licenses/.
> > +#
> > +##############################################################################
> > +
> > +
> > +{
> > -    'name': 'Asterisk Phone Log-call',
> > -    'version': '9.0.0.1.0',
> > -    'category': 'Phone',
> > -    'license': 'AGPL-3',
> > -    'summary': 'Necessary bridge between log-call and Asterisk Click2dial',
> > -    'author': "Trever L. Adams",
> >   No, this is just a suggestion but you might want to add an email in
> >   the README in case at some point there is some claim on copyright. If
> >   we cannot join the authors ("best effort"), it makes things a little
> >   bit more tricky. Again this is a suggestion. Eric CAUDAL
> >   Did I miss spell your name? I am horribly dyslexic. If I did I am very
> >   sorry.

Alright, add it to the README at least. That should be fixed.

Trever

elicoidal commented on asterisk_logcall/models/asterisk_server.py:
```diff
@@ -0,0 +1,64 @@
+        "{odoo_uniqueid} (FreeSWITCH UUID of call).")
+    server_jitter_correction = fields.Integer(
+        string='Time jitter compensation', required=True,
+        help='Number of seconds to subtract from new call start and add to '
```
Thanks for the help messages.
Would it make sense to gather them in the README? (nice to have)

treveradams replied:
On 08/25/2016 05:45 PM, Eric Caudal wrote:

> In asterisk_logcall/models/asterisk_server.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76345099:
> 
> > +
> > +
> > +class AsteriskServer(models.Model):
> > -    _inherit = "asterisk.server"
> >   +
> > -    ucp_url = fields.Char(
> > -        string='Script to download FreeSWITCH recordings', required=False,
> > -        default="https://localhost/cgi-bin/get_recording.pl?file={odoo_filename}",
> > -        help="Macros allowed: {odoo_type} (inbound, outbound), {odoo_src}"
> > -        "(source phone number}, {odoo_dst} (destination number), "
> > -        "{odoo_duration} (length of call), {odoo_start} (start time of call "
> > -        "in seconds since epoch), {odoo_filename} (file name on server), "
> > -        "{odoo_uniqueid} (FreeSWITCH UUID of call).")
> > -    server_jitter_correction = fields.Integer(
> > -        string='Time jitter compensation', required=True,
> > -        help='Number of seconds to subtract from new call start and add to '
> 
> Thanks for the help messages.
> Would it make sense to gather them in the README? (nice to have)
> 
> Should I remove them from the help messages? I have put them into the
> README in an appropriate format.

Trever

elicoidal replied:
I think they are useful in both places actually (people will read the 
README at the beginning but will appreciate to have the message when 
they perform the set up)

treveradams replied:
On 08/26/2016 01:17 AM, Eric Caudal wrote:

> In asterisk_logcall/models/asterisk_server.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76374075:
> 
> > +
> > +
> > +class AsteriskServer(models.Model):
> > -    _inherit = "asterisk.server"
> >   +
> > -    ucp_url = fields.Char(
> > -        string='Script to download FreeSWITCH recordings', required=False,
> > -        default="https://localhost/cgi-bin/get_recording.pl?file={odoo_filename}",
> > -        help="Macros allowed: {odoo_type} (inbound, outbound), {odoo_src}"
> > -        "(source phone number}, {odoo_dst} (destination number), "
> > -        "{odoo_duration} (length of call), {odoo_start} (start time of call "
> > -        "in seconds since epoch), {odoo_filename} (file name on server), "
> > -        "{odoo_uniqueid} (FreeSWITCH UUID of call).")
> > -    server_jitter_correction = fields.Integer(
> > -        string='Time jitter compensation', required=True,
> > -        help='Number of seconds to subtract from new call start and add to '
> >   I think they are useful in both places actually (people will read the
> >   README at the beginning but will appreciate to have the message when
> >   they perform the set up)
> >   I will leave it in both places. Thank you.

Trever

elicoidal commented on freeswitch_click2dial/README.rst:
```diff
@@ -0,0 +1,138 @@
+=====================
+
+The technical name of this module is *freeswitch_click2dial*, but this module
+implements much more than a simple *click2dial* ! This module adds 3
```
remove space before  "!"

elicoidal commented on freeswitch_click2dial/README.rst:
```diff
@@ -0,0 +1,138 @@
+implements much more than a simple *click2dial* ! This module adds 3
+functionalities:
+
+1) It adds a *Dial* button in the partner form view so that users can directly
```
you should probably split:
- summary as introduction here
- detailed operations in usage (with enumerated list and if possible screenshots)
  (nice to have)

elicoidal commented on freeswitch_click2dial/README.rst:
```diff
@@ -0,0 +1,138 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
please adapt

elicoidal commented on freeswitch_click2dial/README.rst:
```diff
@@ -0,0 +1,138 @@
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
remove

elicoidal commented on freeswitch_click2dial/README.rst:
```diff
@@ -0,0 +1,138 @@
+{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
```
remve this line (newest template)

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+from openerp import models, fields, api, _
+from openerp.exceptions import UserError, ValidationError
+import logging
+# from pprint import pformat
```
remove dead code

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+        "leave empty.")
+    password = fields.Char(
+        string='Event Socket Password', required=True,
+        help="Password that OpenERP will use to communicate with the "
```
s/OpenERP/Odoo

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+            alert_info = ('Alert-Info SIP header', server.alert_info)
+            password = ('Event Socket password', server.password)
+
+            if out_prefix[1] and not out_prefix[1].isdigit():
```
would it make sense to create SQL constraints instead?

elicoidal replied:
this is OK for the time being AFAIAC

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+        # We check if the current user has an internal number
+        if not user.resource:
+            raise UserError(
+                _('No resource name configured for the current user'))
```
add a period at the end of the sentences.

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+            raise UserError(
+                _("Problem in the request from Odoo to FreeSWITCH. "
+                  "Here is the error message: %s" % e))
+            # return (False, False, False)
```
remove

elicoidal commented on freeswitch_click2dial/scripts/get_caller_name.py:
```diff
@@ -0,0 +1,284 @@
+
+
+def main(name, phone_number, options):
+    # print 'options = %s' % options
```
remove

elicoidal commented on freeswitch_click2dial/static/src/js/freeswitch_click2dial.js:
```diff
@@ -0,0 +1,96 @@
+        event.stopPropagation();
+        var self = this;
+        self.rpc('/freeswitch_click2dial/get_record_from_my_channel', {}).done(function(r) {
+        // console.log('RESULT RPC r='+r);
```
remove

elicoidal replied:
Sorry again for my lack of precision:
Can you remove dead code from the modules (any commented code in python and jscript modules)?

elicoidal commented on freeswitch_logcall/README.rst:
```diff
@@ -0,0 +1,6 @@
+FreeSWITCH Phone Log-call
```
README template

elicoidal commented on freeswitch_logcall/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+
+{
+    'name': 'FreeSwitch Phone Log-call',
+    'version': '9.0.0.1.0',
```
9.0.1.0.0

elicoidal commented on hr_recruitment_phone_log/README.rst:
```diff
@@ -0,0 +1,44 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/connector-telephony/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/connector-telephony/issues/new?body=module:%20partner_phone_log%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
remove

elicoidal replied:
Sorry, my comment was definitely too short :)
Remove that line as this is not mandatory any more in the newest template (true in all README files.)

elicoidal commented on logcall/__openerp__.py:
```diff
@@ -0,0 +1,41 @@
+
+{
+    'name': 'Phone Log-call',
+    'version': '9.0.0.1.0',
```
9.0.1.0.0

elicoidal commented on logcall/__openerp__.py:
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
```
not sure about the file name: crm_logcall?

treveradams replied:
I don't see crm_logcall. I see a crm_voip_logcall.

elicoidal replied:
I mean that we might want to improve the file name following Odoo recommendations: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules
Starting with the standard base odoo module hence my suggestion (crm_logcall). Nevertheless later I saw there is a crm_voip_logcall and now I am confused about the separation between both.

elicoidal replied:
If this is longstanding module, just let it be not to break any compatibility for migration.

elicoidal commented on logcall/scripts/asterisk_logcall.py:
```diff
@@ -0,0 +1,202 @@
+            odoo = odoorpc.ODOO(options.server, proto, options.port)
+            odoo.login(options.database, options.username, options.password)
+            res = odoo.execute(
+                'phone.common', 'log_call_and_recording', odoo_type, odoo_src, odoo_dst, odoo_duration, odoo_start, odoo_filename, odoo_uniqueid, odoo_description)
```
PEP8

elicoidal commented on logcall/scripts/fusionpbx_get_recording.pl:
```diff
@@ -0,0 +1,72 @@
+    print "Content-Type: text/html\n\n";
+    print "<H1>Error </H1>\n";
+    print "The file ".$args{"file"}." does not exists in the server<br>\n";
+    print "Please contact to the administrator for more info.\n\n";
```
contact the administrator

elicoidal commented on logcall/scripts/fusionpbx_get_recording.pl:
```diff
@@ -0,0 +1,72 @@
+    print "Connection: close\n";
+    print "Content-Type: text/html\n\n";
+    print "<H1>Error </H1>\n";
+    print "The file ".$args{"file"}." does not exists in the server<br>\n";
```
exist

treveradams committed:
Thank you to Alexandr Usov and Stanislav Sinyagin for testing and ideas.
* Fix originate to handle Unicode via UTF-8.
* Fix Open Caller to handle more situations and channel types.
* get_caller_name.py and pop-up functionality now handle invalid numbers properly.
* Other fixes that were already committed.

treveradams commented on hr_recruitment_phone_log/README.rst:
```diff
@@ -0,0 +1,44 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/connector-telephony/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/connector-telephony/issues/new?body=module:%20partner_phone_log%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
On 08/25/2016 05:56 PM, Eric Caudal wrote:

> In hr_recruitment_phone_log/README.rst
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76346020:
> 
> > +Show phonecalls on partner
> > +==========================
> > +
> > +This module restores the phonecall list on the partner view.
> > +
> > +.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
> > -   :alt: Try me on Runbot
> > -   :target: https://runbot.odoo-community.org/runbot/140/9.0
> >   +
> >   +Bug Tracker
> >   +===========
> >   +
> >   +Bugs are tracked on `GitHub Issues <https://github.com/OCA/connector-telephony/issues>`_.
> >   +In case of trouble, please check there if your issue has already been reported.
> >   +If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
> >   +`here <https://github.com/OCA/connector-telephony/issues/new?body=module:%20partner_phone_log%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
> 
> remove
> 
> I am sorry, remove what, the file, the module? I am working to make the
> changes you have requested.

treveradams commented on freeswitch_click2dial/static/src/js/freeswitch_click2dial.js:
```diff
@@ -0,0 +1,96 @@
+        event.stopPropagation();
+        var self = this;
+        self.rpc('/freeswitch_click2dial/get_record_from_my_channel', {}).done(function(r) {
+        // console.log('RESULT RPC r='+r);
```
On 08/25/2016 05:55 PM, Eric Caudal wrote:

> In freeswitch_click2dial/static/src/js/freeswitch_click2dial.js
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76345923:
> 
> > +var click2dial = {};
> > +
> > +click2dial.OpenCaller = Widget.extend({
> > -    template: 'freeswitch_click2dial.OpenCaller',
> >   +
> > -    start: function () {
> > -        this.$('#freeswitch-open-caller').on(
> > -            'click', this.on_open_caller);
> > -        this._super();
> > -    },
> >   +
> > -    on_open_caller: function (event) {
> > -        event.stopPropagation();
> > -        var self = this;
> > -        self.rpc('/freeswitch_click2dial/get_record_from_my_channel', {}).done(function(r) {
> > -        // console.log('RESULT RPC r='+r);
> 
> remove
> 
> Why remove this? It is necessary for FreeSWITCH click to dial
> functionality to work. Yes, it does the same as the Asterisk one, but
> with a different URL.

Trever

treveradams commented on freeswitch_click2dial/freeswitch_click2dial.py:
```diff
@@ -0,0 +1,420 @@
+            alert_info = ('Alert-Info SIP header', server.alert_info)
+            password = ('Event Socket password', server.password)
+
+            if out_prefix[1] and not out_prefix[1].isdigit():
```
On 08/25/2016 05:51 PM, Eric Caudal wrote:

> In freeswitch_click2dial/freeswitch_click2dial.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76345598:
> 
> > -        'res.company', string='Company',
> > -        default=lambda self: self.env['res.company']._company_default_get(
> > -            'freeswitch.server'),
> > -        help="Company who uses the FreeSWITCH server.")
> >   +
> > -    @api.multi
> > -    @api.constrains(
> > -        'out_prefix', 'wait_time', 'port', 'context', 'alert_info', 'password')
> > -    def _check_validity(self):
> > -        for server in self:
> > -            out_prefix = ('Out prefix', server.out_prefix)
> > -            dialplan_context = ('Dialplan context', server.context)
> > -            alert_info = ('Alert-Info SIP header', server.alert_info)
> > -            password = ('Event Socket password', server.password)
> >   +
> > -            if out_prefix[1] and not out_prefix[1].isdigit():
> 
> would it make sense to create SQL constraints instead?
> 
> Maybe. I am new to OpenERP/Odoo. I am basing much of this off of the
> Asterisk work in connector-telephony. Can you give me pointers?

In a moment, I will have addressed everything but those that I have
asked questions about (including some of those actually).

Trever

elicoidal replied:
some pointers here: 
http://www.odoo.com/documentation/9.0/howtos/backend.html#model-constraints

treveradams replied:
On 08/26/2016 01:20 AM, Eric Caudal wrote:

> In freeswitch_click2dial/freeswitch_click2dial.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76374295:
> 
> > -        'res.company', string='Company',
> > -        default=lambda self: self.env['res.company']._company_default_get(
> > -            'freeswitch.server'),
> > -        help="Company who uses the FreeSWITCH server.")
> >   +
> > -    @api.multi
> > -    @api.constrains(
> > -        'out_prefix', 'wait_time', 'port', 'context', 'alert_info', 'password')
> > -    def _check_validity(self):
> > -        for server in self:
> > -            out_prefix = ('Out prefix', server.out_prefix)
> > -            dialplan_context = ('Dialplan context', server.context)
> > -            alert_info = ('Alert-Info SIP header', server.alert_info)
> > -            password = ('Event Socket password', server.password)
> >   +
> > -            if out_prefix[1] and not out_prefix[1].isdigit():
> >   some pointers here:
> >   http://www.odoo.com/documentation/9.0/howtos/backend.html#model-constraints
> 
> Alright. Thank you for all of your help. This is the only suggested
> change that I have not yet made. (Other than a not working -> TODO
> change, which I explained).

I will look at this one later today.

Thank you.
Trever

treveradams commented:
The rebase should have addressed the rest of your concerns unless I was needing clarification.

elicoidal commented on asterisk_logcall/README.rst:
```diff
@@ -0,0 +1,75 @@
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
```
I think I miss part of the standard footer. Can you check it again?

elicoidal replied:
no pb!

elicoidal commented on crm_voip_logcall/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
```
short headers

elicoidal replied:
(all py files)

elicoidal commented on freeswitch_click2dial/freeswitch_click2dial_demo.xml:
```diff
@@ -0,0 +1,39 @@
+<data noupdate="1">
+
+<record id="demo_fs_server" model="freeswitch.server">
+    <field name="name">Akretion FreeSWITCH Server</field>
```
s/Akretion/Company

elicoidal commented on freeswitch_click2dial/static/src/js/freeswitch_click2dial.js:
```diff
@@ -0,0 +1,93 @@
+
+            }
+        else if (typeof r == 'object' && r.length == 3) {
+            self.do_notify( // Not working
```
if not working maybe add a //TODO or FIXME?

elicoidal replied:
ok

elicoidal commented on freeswitch_logcall/README.rst:
```diff
@@ -0,0 +1,75 @@
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
```
Check all README footer as something disappeared here...

elicoidal commented:
@treveradams good improvements. Some small details on my side but getting somewhere.
I did not test it and I am not qualified for deep code review.
@alexis-via @pedrobaeza might help here on the technical side.

treveradams commented on asterisk_logcall/README.rst:
```diff
@@ -0,0 +1,75 @@
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
```
On 08/26/2016 01:22 AM, Eric Caudal wrote:

> In asterisk_logcall/README.rst
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76374465:
> 
> > +
> > +\* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
> > +
> > +Contributors
> > +------------
> > +
> > +\* Trever L. Adams
> > +
> > +Maintainer
> > +----------
> > +
> > +.. image:: https://odoo-community.org/logo.png
> > -   :alt: Odoo Community Association
> > -   :target: https://odoo-community.org
> >   +
> >   +This module is maintained by the OCA.
> 
> I think I miss part of the standard footer. Can you check it again?
> 
> I am not sure how that happened. I messed up a copy and paste somehow. I
> believe I have found and fixed all the places that happened.

Trever

treveradams commented on logcall/__openerp__.py:
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
```
On 08/26/2016 01:32 AM, Eric Caudal wrote:

> In logcall/**openerp**.py
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76375612:
> 
> > @@ -0,0 +1,41 @@
> > +# -_\- coding: utf-8 -_-
> 
> I mean that we might want to improve the file name following Odoo
> recommendations:
> https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules
> Starting with the standard base odoo module hence my suggestion
> (crm_logcall). Nevertheless later I saw there is a crm_voip_logcall
> and now I am confused about the separation between both.
> 
> The logcall module was originally written by credative.co.uk. I believe
> there is a crm_voip module in 8.0 if the module has been purchased and
> part of 9.0 enterprise. This module is simply a way to bridge the two.

Trever

treveradams commented on freeswitch_click2dial/static/src/js/freeswitch_click2dial.js:
```diff
@@ -0,0 +1,93 @@
+
+            }
+        else if (typeof r == 'object' && r.length == 3) {
+            self.do_notify( // Not working
```
On 08/26/2016 01:28 AM, Eric Caudal wrote:

> In freeswitch_click2dial/static/src/js/freeswitch_click2dial.js
> https://github.com/OCA/connector-telephony/pull/101#discussion_r76375235:
> 
> > -        }
> > -        else if (typeof r == 'string') {
> > -             var action = {
> > -                name: _t('Number Not Found'),
> > -                type: 'ir.actions.act_window',
> > -                res_model: 'number.not.found',
> > -                view_mode: 'form',
> > -                views: [[false, 'form']],
> > -                target: 'new',
> > -                context: {'default_calling_number': r},
> > -             };
> > -            web_client.action_manager.do_action(action);
> >   +
> > -            }
> > -        else if (typeof r == 'object' && r.length == 3) {
> > -            self.do_notify( // Not working
> 
> if not working maybe add a //TODO?
> 
> This is code that is a direct copy from asterisk_click2dial with just
> the renames. I would prefer not to change anything on a line that is
> functional as it will make it harder to keep in sync.

Trever

treveradams committed:
Do much of the cleanup work requested by Eric Caudal.
Remove an accidentally left in rebase problem.

treveradams committed:
Cleanup requested by Eric Caudal.

treveradams commented:
On 08/26/2016 01:36 AM, Eric Caudal wrote:

> @treveradams https://github.com/treveradams good improvements. Some
> small details on my side but getting somewhere.
> I did not test it and I am not qualified for deep code review.
> @alexis-via https://github.com/alexis-via @pedrobaeza
> https://github.com/pedrobaeza might help here on the technical side.
> 
> Alexandr Usov did testing for FreeSWITCH-8.0 (I do not have a working
> logcall-8.0 yet).

Of course, any deep code review and testing would be greatly appreciated
(by whomever does it).

Trever

treveradams commented:
Please, hold off on the review. I mistakenly missed changing to short copyright info at the top of the files in freeswitch_click2dial. Additionally, Alexandr Usov just found a problem with Open Caller that I would like to fix before this gets merged. Hopefully that can be addressed in the next hour or two.

treveradams committed:
Fix bug introduced by dealing with invalid numbers.
Cleanup file copyrights as suggested by Eric Caudal.
get_caller_name.py should handle Unicode results properly.

treveradams commented:
Now ready for review.
Ok, thank you again to Alexandr Usov for helping to test. We found a problem with Open Caller that would cause a traceback in python if the number wasn't provided (empty, unknown, etc.). We fixed that, but caused Open Caller not to work. This has been fixed. I believe this module is now ready for review. It has been rebased onto the said work in freeswitch-9.0 (the version is Odoo version). The freeswitch-7.0 and 8.0 branches have the same corrections. logcall-8.0 exists but doesn't work now. I am open to help there if people are interested. Unless a 7.0 port of logcall is straightforward from the 8.0, I won't even consider that.

treveradams committed:
More PEP8 related cleanups.

treveradams committed:
More PEP8 related cleanups.

treveradams committed:
More PEP8 related cleanups.
